test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs

```diff
@@ -864,6 +864,224 @@ public void ObservableCounterSpatialAggregationTest(bool exportDelta)
             Assert.Equal(30, metricPoint1.GetSumLong());
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void UpDownCounterAggregationTest(bool exportDelta)
+        {
+            DateTime testStartTime = DateTime.UtcNow;
+
+            var exportedItems = new List<Metric>();
+
+            using var meter = new Meter($"{Utils.GetCurrentMethodName()}.{exportDelta}");
+            var counterLong = meter.CreateUpDownCounter<long>("mycounter");
+            using var meterProvider = Sdk.CreateMeterProviderBuilder()
+                .AddMeter(meter.Name)
+                .AddInMemoryExporter(exportedItems, metricReaderOptions =>
+                {
+                    metricReaderOptions.TemporalityPreference = exportDelta ? MetricReaderTemporalityPreference.Delta : MetricReaderTemporalityPreference.Cumulative;
+                })
+                .Build();
+
+            counterLong.Add(10);
+            counterLong.Add(-5);
+            meterProvider.ForceFlush(MaxTimeToAllowForFlush);
+            long sumReceived = GetLongSum(exportedItems);
+            Assert.Equal(5, sumReceived);
+
+            var metricPoint = GetFirstMetricPoint(exportedItems);
+            Assert.NotNull(metricPoint);
+            Assert.True(metricPoint.Value.StartTime >= testStartTime);
+            Assert.True(metricPoint.Value.EndTime != default);
+
+            DateTimeOffset firstRunStartTime = metricPoint.Value.StartTime;
+            DateTimeOffset firstRunEndTime = metricPoint.Value.EndTime;
+
+            exportedItems.Clear();
+
+#if NETFRAMEWORK
+            Thread.Sleep(5000); // Compensates for low resolution timing in netfx.
```

Comment:
Curious, why `5000` (this is a very long time from the CPU's perspective)?

Reply:
CPUs always want things right now! They oughta learn to have some patience.

But, yea you're right... I decreased it and also decreased it from the test I borrowed this from.